### PR TITLE
Support GraphGym custom modules outside PyG package

### PR DESCRIPTION
**Pull request by JiaxuanYou**
List of changes:
1. Create `graphgym/custom_graphgym` directory outside Kumo package. 
**Users can register their customized modules here, without touching Kumo package.** 
To achieve this, I remove all the `torch_geometric.graphgym.contrib` calls within our pacakge; instead, I import these modules in `graphgym/main.py`.
2. Keep `torch_geometric/graphgym/contrib`, since some functionalities are needed. **This could be the place where advanced users PR to our repo**. This could be refactored directly as `torch_geometric/contrib` as well.
3. Fix bugs so that GraphGym can work with our current fork. See Issue #101. Things have been fixed now.

### CODE REVIEW

**Review by rusty1s — COMMENTED** [review]
Thanks for the clean-up @JiaxuanYou. This looks great:

* Can you say a bit more on why we still need the `contrib` package in `torch_geometric.graphgym`? It looks like most folders are empty besides `layer` (and we already have a `GeneralConv` in `torch_geometric.nn.conv`).

File: graphgym/custom_graphgym/act/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from os.path import dirname, basename, isfile, join
+import glob
+
+modules = glob.glob(join(dirname(__file__), "*.py"))
```
[the remark this answers is not in the document]
Comment (JiaxuanYou) [reply graphgym/custom_graphgym/act/__init__.py:4]: > Thanks for the clean-up @JiaxuanYou. This looks great:
> 
> * Can you say a bit more on why we still need the `contrib` package in `torch_geometric.graphgym`? It looks like most folders are empty besides `layer` (and we already have a `GeneralConv` in `torch_geometric.nn.conv`).

I agree it might look a bit weird to have them empty.
My intention is that we can keep customized/unofficial modules in 2 places: within and outside Kumo.
The current `contrib` is more like a suggested marketplace, where people and our team can PR into it. Besides, the nice thing is that you can directly copy paste the files in `custom_graphgym` into the `contrib` dir and that is. Otherwise, it's less clear for people to contribute to our package if needed.


I'll try not including `__all__` to see if it works. I hacked this solution a while ago and it has been working well, so I didn't change it afterwards :)

[the remark this answers is not in the document]
Comment (rusty1s) [reply graphgym/custom_graphgym/act/__init__.py:4]: Thanks. Besides in `contrib`, where do other ops get registered, e.g., `GCNConv`, `ReLU`, `BatchNorm`, ...? I can't seem to find it.

Yes, please check if empty `__init__.py` works as well :)

[the remark this answers is not in the document]
Comment (JiaxuanYou) [reply graphgym/custom_graphgym/act/__init__.py:4]: Unfortunately, I found that `__all__` has to be defined in order to automatically register all the modules in the directory.

[the remark this answers is not in the document]
Comment (JiaxuanYou) [reply graphgym/custom_graphgym/act/__init__.py:4]: > Thanks. Besides in contrib, where do other ops get registered, e.g., GCNConv, ReLU, BatchNorm, ...? I can't seem to find it.

These modules are defined [here](https://github.com/vrtex-team/pytorch_geometric/blob/master/torch_geometric/graphgym/models/layer.py), since they are some commonly-used building blocks.

File: graphgym/main.py [hunk 2]
```diff
@@ -4,6 +4,9 @@
 import torch
 import logging
 
+import torch_geometric.graphgym.contrib # noqa, register contrib modules
```
[the remark this answers is not in the document]
Comment (JiaxuanYou) [reply graphgym/main.py:7]: I move this line to `torch_geometric.graphgym.__init__.py` and it's no longer needed. Thanks
